### ISSUE TITLE
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'repository' of null

### ISSUE THREAD

**Issue by wpschaub** [rfennell/AzurePipelines#752]
**Describe the bug**
Generate Release Notes fails with error `UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'repository' of null`, which results in **no** markdown file being generated and Git based Wiki Updater failing the pipeline.

**Logs**

```
2020-06-26T01:39:34.5761855Z ##[section]Starting: Generate Release Notes based on Release Comparison API
2020-06-26T01:39:34.5948173Z ==============================================================================
2020-06-26T01:39:34.5948722Z Task         : Generate Release Notes (Crossplatform)
2020-06-26T01:39:34.5949284Z Description  : Generates a release notes file in a format of your choice from the build or release history
2020-06-26T01:39:34.5949757Z Version      : 2.52.2
2020-06-26T01:39:34.5950093Z Author       : Black Marble
2020-06-26T01:39:34.5951896Z Help         : Version: 2.52.2. [More Information](https://github.com/rfennell/vNextBuild/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task/)
2020-06-26T01:39:34.5952814Z ==============================================================================
2020-06-26T01:39:35.4196104Z Getting the current release details
2020-06-26T01:39:35.7862206Z Show associated items for primary artifact only is set to false
2020-06-26T01:39:35.7863287Z Getting deployment count for stage
2020-06-26T01:39:35.7864189Z Identified [pr] as having deployment count of [1]
2020-06-26T01:39:35.7866570Z Getting the Environment Id
2020-06-26T01:39:35.7867244Z Identified [pr] as having id [620]
2020-06-26T01:39:35.8586340Z Getting all artifacts in the current release...
2020-06-26T01:39:35.8591323Z Found 2
2020-06-26T01:39:35.9979403Z Getting all artifacts in the most recent successful release [Release-20]...
2020-06-26T01:39:35.9980348Z Found 2
2020-06-26T01:39:35.9980917Z Looking at artifact [SensorService]
2020-06-26T01:39:35.9981476Z Artifact type [Build]
2020-06-26T01:39:35.9982056Z Build Definition ID [1337]
2020-06-26T01:39:35.9983421Z Build Number: [341840]
2020-06-26T01:39:35.9983955Z Is Primary: [true]
2020-06-26T01:39:35.9984568Z Looking for the [SensorService] in the most recent successful release [Release-20]
2020-06-26T01:39:35.9985510Z Found artifact [SensorService] with build number [301056] in release [Release-20]
2020-06-26T01:39:35.9986554Z Checking what commits and workitems have changed from [301056][ID 301056] => [341840] [ID 341840]
2020-06-26T01:39:35.9987592Z Defaulting on the workaround for build API limitation (see issue #349 set 'ReleaseNotes.Fix349=false' to disable)
2020-06-26T01:39:36.0957101Z Using workaround for build API limitation (see issue #349)
2020-06-26T01:39:36.2942458Z Enriched change 9a1da3f1bd082c46a1bbd3cea095dc403906a1a2 of type TfsGit
2020-06-26T01:39:36.5772097Z Enriched with details of 6 files
2020-06-26T01:39:36.5773308Z Enriched change 623fbf336649828be37ce4e593f9bc986fde017a of type TfsGit
2020-06-26T01:39:36.6472220Z Enriched with details of 17 files
2020-06-26T01:39:36.6473160Z Enriched change 872013068c131cc21d84b5945c67911ccdd4913d of type TfsGit
2020-06-26T01:39:36.8192204Z Enriched with details of 2 files
2020-06-26T01:39:36.8193347Z Enriched change 9198b2a2b80c87532201feaea5a54e8c5ec664b8 of type TfsGit
2020-06-26T01:39:36.8699261Z Enriched with details of 14 files
2020-06-26T01:39:36.8700156Z Enriched change 8ff061c73e9e0be56a2dc66ae55c6caa2e531082 of type TfsGit
2020-06-26T01:39:36.9144827Z Enriched with details of 2 files
2020-06-26T01:39:36.9145825Z Enriched change 654b7c52e590e18f74b6a52df9cda0deec5aa1dc of type TfsGit
2020-06-26T01:39:36.9640273Z Enriched with details of 6 files
2020-06-26T01:39:36.9641566Z Enriched change 3d77168394ab1443c90fc48d84c02cd2138323e9 of type TfsGit
2020-06-26T01:39:37.0111699Z Enriched with details of 10 files
2020-06-26T01:39:37.0112697Z Enriched change 237cb3a6e9f552183f6d4714f2e522cf03e047d0 of type TfsGit
2020-06-26T01:39:37.3930739Z Enriched with details of 2 files
2020-06-26T01:39:37.3931869Z Enriched change f30f1fa22b8e2f2e5a380dd6f6c1a021c8eeb129 of type TfsGit
2020-06-26T01:39:37.4388632Z Enriched with details of 10 files
2020-06-26T01:39:37.4389690Z Enriched change d544b0381de7b3022679b0eefe30459a906e888d of type TfsGit
2020-06-26T01:39:37.5816575Z Enriched with details of 5 files
2020-06-26T01:39:37.5818113Z Enriched change d3a042432830ebabc2c76a3c8fd4b96320ab9bab of type TfsGit
2020-06-26T01:39:37.6313185Z Enriched with details of 3 files
2020-06-26T01:39:37.6317896Z Enriched change c5584166c2fae36610f236a3bb60aa312cf9d244 of type TfsGit
2020-06-26T01:39:37.6840427Z Enriched with details of 9 files
2020-06-26T01:39:37.6841262Z Enriched change a84ab19a6acc9d49fe78099086a740b0535aabdf of type TfsGit
2020-06-26T01:39:37.7284635Z Enriched with details of 9 files
2020-06-26T01:39:37.7285847Z Enriched change 86808b1b46158b8c4561d9fcd8cd9998a9d45316 of type TfsGit
2020-06-26T01:39:37.8025469Z Enriched with details of 111 files
2020-06-26T01:39:37.8026572Z Enriched change e03ee676f8d87edaf2ae36a2b42761de72233ac0 of type TfsGit
2020-06-26T01:39:37.8522423Z Enriched with details of 11 files
2020-06-26T01:39:37.8523316Z Enriched change 7ae20d6c8ca1da2640c7ab894b92e0d4b2dd7c9f of type TfsGit
2020-06-26T01:39:37.8954467Z Enriched with details of 2 files
2020-06-26T01:39:37.8955571Z Enriched change 58b32d56dd84f0b8dd6fda2092f4d00708d0d251 of type TfsGit
2020-06-26T01:39:37.9413538Z Enriched with details of 17 files
2020-06-26T01:39:37.9414531Z Getting test associated with the latest build [341840]
2020-06-26T01:39:38.2219009Z Found 65 test associated with the build [341840] adding any not already in the global test list to the list
2020-06-26T01:39:38.2906850Z Adding the build [341840] and its association to the unified results object
2020-06-26T01:39:38.2908091Z Get details of [31] WIs
2020-06-26T01:39:38.2908801Z Getting full details of WI batch from index: [0] to [31]
2020-06-26T01:39:38.8894069Z Adding [16] items
2020-06-26T01:39:38.8896018Z Detected 17 commits/changesets and 31 workitems between the current build and the last successful one
2020-06-26T01:39:38.8897264Z Detected 65 tests associated within the current build.
2020-06-26T01:39:38.8897784Z 
2020-06-26T01:39:38.8898470Z Looking at artifact [Infrastructure]
2020-06-26T01:39:38.8899251Z Artifact type [Build]
2020-06-26T01:39:38.8899877Z Build Definition ID [701]
2020-06-26T01:39:38.8900519Z Build Number: [336629]
2020-06-26T01:39:38.8901044Z Is Primary: [undefined]
2020-06-26T01:39:38.8901964Z Looking for the [Infrastructure] in the most recent successful release [Release-20]
2020-06-26T01:39:38.8903514Z Found artifact [Infrastructure] with build number [182574] in release [Release-20]
2020-06-26T01:39:38.8904572Z Checking what commits and workitems have changed from [182574][ID 182574] => [336629] [ID 336629]
2020-06-26T01:39:38.9359611Z Using workaround for build API limitation (see issue #349)
2020-06-26T01:39:38.9395595Z (node:12880) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'repository' of null
2020-06-26T01:39:38.9524083Z ##[section]Finishing: Generate Release Notes based on Release Comparison API
```

**What is the issue see:**
 - Azure DevOps Services (VSTS)

**Comment by wpschaub** [comment]
Here is an extract from the log with system.debug = true:

```
...
2020-06-26T13:10:01.7652156Z Total Pull Requests: [8]
2020-06-26T13:10:01.8058948Z ##[debug]Using template mode File
2020-06-26T13:10:01.8073264Z Loading template file d:\_work1\r180\a\ReleaseNotesTemplate.mk
2020-06-26T13:10:01.8083590Z ##[debug]Loading handlebar template
2020-06-26T13:10:01.8121905Z ##[debug]Processing template
2020-06-26T13:10:01.8123033Z ##[debug]  WI: 66
2020-06-26T13:10:01.8123795Z ##[debug]  CS: 99
2020-06-26T13:10:01.8124485Z ##[debug]  PR: 8
2020-06-26T13:10:01.8125143Z ##[debug]  Builds: 2
2020-06-26T13:10:01.8125957Z ##[debug]  Global Tests: 65
2020-06-26T13:10:01.8126673Z ##[debug]  Release Tests: 0
2020-06-26T13:10:01.8127431Z ##[debug]  Related WI: 0
2020-06-26T13:10:01.8128189Z ##[debug]Processing handlebar template
2020-06-26T13:10:01.8878102Z Loading handlebars-helpers extension
2020-06-26T13:10:02.3570770Z ##[debug]Saving custom Handlebars code to file in folder d:\_work1/_temp
2020-06-26T13:10:02.4876251Z Completed processing template
2020-06-26T13:10:02.4877963Z Writing output file d:\_work1\r180\a/ReleaseNotes.md
2020-06-26T13:10:02.5052000Z Finished writing output file d:\_work1\r180\a/ReleaseNotes.md
2020-06-26T13:10:02.5053075Z ##[debug]task result: Succeeded
2020-06-26T13:10:02.5056640Z ##[debug]Processed: ##vso[task.complete result=Succeeded;]
2020-06-26T13:10:02.5293058Z ##[section]Finishing: Generate Release Notes based on Release Comparison API
```

**Comment by wpschaub** [comment]
And here is the template we are using:

**Comment by wpschaub** [comment]
@rfennell I have not been able to reproduce the error. Any way we can catch these unhandled exceptions as well?

**Comment by rfennell** [comment]
I think another try/catch to continue on is in order, I will get it done

**Comment by rfennell** [comment]
This has been released in 2.53.2

**Comment by wpschaub** [comment]
Thank you @rfennell - have a great weekend and I will advise if we pickup another issue :)
